_content/tour/rus/goroutines.article

```diff
@@ -1,116 +1,114 @@
-of cores that are available to the program. For each thread that can be run in parallel,
-the runtime creates an operating system thread (M) and attaches that to a data structure
-that represents a logical processor (P) inside the program. This P and M represent the
-compute power or execution context for running the Go program.
+При запуске Go-программы среда выполнения Go спрашивает машину (виртуальную или физическую)
+сколько потоков операционной системы может работать параллельно. Этот вопрос основывается на количестве
+ядер, доступных программе. Для каждого потока, который может работать параллельно,
+среда выполнения создает поток операционной системы (M) и прикрепляет его к структуре данных
+которая представляет собой логический процессор (P) внутри программы. Эти P и M представляют собой
+вычислительные мощности или контекст исполнения для выполнения программы Go.
 
-Also, an initial Goroutine (G) is created to manage the execution of instructions
-on a selected M/P. Just like an M manages the execution of instructions on the hardware,
-a G manages the execution of instructions on the M. This creates a new layer of
-abstraction above the operating system, but it moves execution control to the
-application level.
+Также создается начальная горутина (G) для управления выполнением инструкций
+на выбранном M/P. Точно так же, как M управляет выполнением инструкций на аппаратном обеспечении,
+G управляет выполнением инструкций на М. Это создает новый уровень
+абстракции над операционной системой, но переносит управление выполнением на
+уровень приложений.
 
 .image /tour/eng/static/img/gor1.png
 
-Since the Go scheduler sits on top of the operating system scheduler, it’s important
-to have some semantic understanding of the operating system scheduler and the constraints
-it applies to the Go scheduler and applications.
+Поскольку планировщик Go располагается поверх планировщика операционной системы, важно
+иметь некоторое семантическое представление о планировщике операционной системы и ограничениях.
+которые он накладывает на планировщик Go и приложения.
 
-The operating system scheduler has the job of creating the illusions that multiple
-pieces of work are being executed at the same time. Even when this is physically
-impossible. This requires some tradeoffs in the design of the scheduler. Before I
-go any further, it’s important to define some words.
+Задача планировщика операционной системы - создавать иллюзию того, что несколько
+частей работы выполняются одновременно. Даже если это физически
+невозможно. Это требует некоторых компромиссов при проектировании планировщика. Прежде чем 
+продолжить, важно дать определение некоторым словам.
 
-*Work:* A set of instructions to be executed for a running application. This is
-accomplished by threads and an application can have 1 to many threads.
+*Работа:* Набор инструкций для выполнения в выполняемом приложении. Это
```

Comment:
```suggestion
*Работа:* Набор инструкций для выполнения в запущенном приложении. Это
```

just to avoid words with the same root next to each other